Description: Disable csd-printer plugin under Unity. We are still using system-config-printer for this, so no need to start another service for this.
Author: Martin Pitt <martin.pitt@ubuntu.com>
Forwarded: No, Ubuntu specific

Index: cinnamon-settings-daemon-3.6.1/plugins/print-notifications/csd-print-notifications-plugin.c
===================================================================
--- cinnamon-settings-daemon-3.6.1.orig/plugins/print-notifications/csd-print-notifications-plugin.c	2012-10-05 13:56:44.000000000 +0200
+++ cinnamon-settings-daemon-3.6.1/plugins/print-notifications/csd-print-notifications-plugin.c	2012-10-26 10:19:49.611004182 +0200
@@ -38,6 +38,12 @@
 static void
 csd_print_notifications_plugin_init (CsdPrintNotificationsPlugin *plugin)
 {
+        if (g_strcmp0 (g_getenv ("XDG_CURRENT_DESKTOP"), "Unity") == 0) {
+            plugin->priv = NULL;
+            g_debug ("CsdPrintNotificationsPlugin: Disabling for Unity, using system-config-printer");
+            return;
+        }
+
         plugin->priv = CSD_PRINT_NOTIFICATIONS_PLUGIN_GET_PRIVATE (plugin);
 
         plugin->priv->manager = csd_print_notifications_manager_new ();
@@ -55,6 +61,11 @@
 
         plugin = CSD_PRINT_NOTIFICATIONS_PLUGIN (object);
 
+        if (g_strcmp0 (g_getenv ("XDG_CURRENT_DESKTOP"), "Unity") == 0) {
+            G_OBJECT_CLASS (csd_print_notifications_plugin_parent_class)->finalize (object);
+            return;
+        }
+
         g_return_if_fail (plugin->priv != NULL);
 
         if (plugin->priv->manager != NULL) {
@@ -70,6 +81,11 @@
         gboolean res;
         GError  *error;
 
+        if (CSD_PRINT_NOTIFICATIONS_PLUGIN (plugin)->priv == NULL) {
+            g_debug ("Not activating disabled print-notifications plugin");
+            return;
+        }
+
         g_debug ("Activating print-notifications plugin");
 
         error = NULL;
@@ -83,6 +99,11 @@
 static void
 impl_deactivate (CinnamonSettingsPlugin *plugin)
 {
+        if (CSD_PRINT_NOTIFICATIONS_PLUGIN (plugin)->priv == NULL) {
+            g_debug ("Not deactivating disabled print-notifications plugin");
+            return;
+        }
+
         g_debug ("Deactivating print_notifications plugin");
         csd_print_notifications_manager_stop (CSD_PRINT_NOTIFICATIONS_PLUGIN (plugin)->priv->manager);
 }
